File: server/src/main/java/com/pass/exam/ui/ExamWebConfig.java
package com.pass.exam.ui;

import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
@RequiredArgsConstructor
public class ExamWebConfig implements WebMvcConfigurer {

    private final ExamStatusConverter examStatusConverter;

    @Override
    public void addFormatters(org.springframework.format.FormatterRegistry registry) {
        registry.addConverter(examStatusConverter);
    }
}

File: server/src/main/kotlin/com/fluffy/exam/ui/ExamWebConfig.kt
package com.fluffy.exam.ui

import org.springframework.context.annotation.Configuration
import org.springframework.format.FormatterRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class ExamWebConfig(
    private val examStatusConverter: ExamStatusConverter
) : WebMvcConfigurer {

    override fun addFormatters(registry: FormatterRegistry) {
        registry.addConverter(examStatusConverter)
    }
}